lib/PostEditorBlocks/WooCommerceBlocksIntegration.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace MailPoet\PostEditorBlocks;
+
+use Automattic\WooCommerce\Blocks\Domain\Services\ExtendRestApi;
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\StoreApi\Schemas\CheckoutSchema;
+use MailPoet\Config\Env;
+use MailPoet\Models\Subscriber;
+use MailPoet\Settings\SettingsController;
+use MailPoet\WooCommerce\Subscription as WooCommerceSubscription;
+use MailPoet\WP\Functions as WPFunctions;
+
+// @todo Disable if blocks is inactive
+class WooCommerceBlocksIntegration {
+  /** @var SettingsController */
+  private $settings;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    WPFunctions $wp,
+    SettingsController $settings,
+    WooCommerceSubscription $woocommerceSubscription
+  ) {
+    $this->wp = $wp;
+    $this->settings = $settings;
+    $this->woocommerceSubscription = $woocommerceSubscription;
+  }
+
+  public function init() {
+    $this->wp->addAction(
+      'woocommerce_blocks_checkout_block_registration',
+      [$this, 'registerCheckoutFrontendBlocks']
+    );
+    $this->wp->addAction(
+      '__experimental_woocommerce_blocks_checkout_update_order_from_request',
```

Comment:
I can see this experimental Hook. What happens when the hook will be moved from the experimental status? Will you create a new pull request? Would it work for all WooCommerce versions?

Reply:
If the __experimental hook was graduated to non-experimental in the Blocks plugin, deprecation would be added and this would need changing. I won't be creating new pull requests for this integration. It is my understanding that this work will be handed off to the MailPoet team once complete.

Reply:
Just to confirm that once we graduate this, `__experimental_woocommerce_blocks_checkout_update_order_from_request` would still function just as fine and the deprecation warning would outline any changes you might need to do.
We consider that hook fairly stable so far, so I doubt we will need to change anything.

Reply:
@senadir looks like it was graduated already so let's swap.